qcore/im.py

```diff
@@ -99,3 +117,45 @@ def order_ims(unsorted_ims, pattern_order=DEFAULT_PATTERN_ORDER):
         [adj_ims.append(im) for im in unsorted_ims if im not in adj_ims]
 
     return adj_ims
+
+
+@dataclass
+class IM:
+    name: IMEnum
+    period: int = None
+    component: constants.Components = None
+
+    def get_im_name(self):
+        if self.period:
+            return f"{self.name}_{self.period}"
+        else:
+            return self.name
+
+    def pretty_im_name(self):
+        """
+        :return: IM name in the form "IM_NAME [UNITS]" or "IM_NAME (PERIOD|UNITS) [UNITS]"
+        """
+        if self.period:
+            return f"{self.name} ({self.period}{self.get_period_unit()}) [{self.get_unit()}]"
+        else:
+            return f"{self.name} [{self.get_unit()}]"
+
+    def get_unit(self):
+        if self.name in [IMEnum.PGA.name, IMEnum.pSA.name]:
```

Comment:
self.name is an IMEnum. IMEnum.PGA.name is a str.

Reply:
Unless this actually works, in which case having strings was fine

Reply:
Good catch. Is there a good way to extend init to covert name from a str to IMEnum? (I've accidentally used it as a string in my EE script?

Reply:
IMEnum('name') should return the right enum member. Otherwise you have to use IMEnum.from_str('str_value') if IMEnum is an ExtendedStrEnum

Reply:
e.g. Component('c000') and Component.from_str('000') give the same enum member

Reply:
I'm glad to see this code is well tested :P